src/pages/billing/Project.js

```diff
@@ -14,187 +14,176 @@ import * as StateHistory from 'src/libs/state-history'
-      }),
-      !!deletingUser && h(DeleteUserModal, {
-        userEmail: deletingUser.email,
-        onDismiss: () => this.setState({ deletingUser: false }),
-        onSubmit: _.flow(
-          withErrorReporting('Error removing member from billing project'),
-          Utils.withBusyState(v => this.setState({ updating: v }))
-        )(async () => {
-          this.setState({ deletingUser: false })
-          await Ajax().Billing.project(projectName).removeUser(deletingUser.roles, deletingUser.email)
-          this.refresh()
-        })
-      }),
-      (refreshing || loadingBillingInfo || updatingAccount || updating) && spinnerOverlay
-    ])
-  }
-
-  componentDidUpdate(prevProps, prevState) {
-    const hasBillingScope = Auth.hasBillingScope()
-    if (prevState.hasBillingScope !== hasBillingScope) {
-      this.setState({ hasBillingScope })
-      hasBillingScope && Utils.withBusyState(loading => this.setState({ loading }), this.loadBillingInfo)()
-    }
+      div({
+        style: {
+          marginTop: '1rem',
+          display: 'flex'
+        }
+      }, [
+        h(NewUserCard, {
+          onClick: () => setAddingUser(true)
+        }),
+        div({ style: { flexGrow: 1 } },
+          _.map(member => {
+            return h(MemberCard, {
+              adminLabel: 'Owner',
+              userLabel: 'User',
+              member, adminCanEdit,
+              onEdit: () => setEditingUser(member),
+              onDelete: () => setDeletingUser(member)
+            })
+          }, projectUsers)
+        )
+      ])
+    ]),
+    addingUser && h(NewUserModal, {
+      adminLabel: 'Owner',
+      userLabel: 'User',
+      title: 'Add user to Billing Project',
+      footer: 'Warning: Adding any user to this project will mean they can incur costs to the billing associated with this project.',
+      addFunction: Ajax().Billing.project(projectName).addUser,
+      onDismiss: () => setAddingUser(false),
+      onSuccess: refresh
+    }),
+    editingUser && h(EditUserModal, {
+      adminLabel: 'Owner',
+      userLabel: 'User',
+      user: editingUser,
+      saveFunction: Ajax().Billing.project(projectName).changeUserRoles,
+      onDismiss: () => setEditingUser(false),
+      onSuccess: refresh
+    }),
+    !!deletingUser && h(DeleteUserModal, {
+      userEmail: deletingUser.email,
+      onDismiss: () => setDeletingUser(false),
+      onSubmit: _.flow(
+        withErrorReporting('Error removing member from billing project'),
+        Utils.withBusyState(updating)
+      )(async () => {
+        setDeletingUser(false)
+        await Ajax().Billing.project(projectName).removeUser(deletingUser.roles, deletingUser.email)
+        refresh()
+      })
+    }),
+    (refreshing || loadingBillingInfo || updatingAccount || updating) && spinnerOverlay
+  ])
+}
 
-    StateHistory.update(_.pick(
-      ['projectUsers', 'filter'],
-      this.state)
-    )
-  }
-})
+export default ProjectDetail
```

Comment:
What utility does it provide to export the default at the bottom rather than inlining it?

Reply:
We actually require it with an eslint rule.

Specifically for a default, you can't name it if you export it inline and it's not a class:

```js
export default () => {} // this works, but no name associated with it, e.g. in the React dev tools*
export default class Foo extends Component {} // this works, but it's a class
const Bar = () => {}
export default Bar // this works and has a name!

export default const Baz = () => {} // not a thing, unfortunately
````

* any component can have a name associated with it that'll show up in the dev tools manually, I made `Utils.withDisplayName` to make this easier, but it's still an extra step

Reply:
This is a great explanation - is there someplace we could save this for future reference?